globus_cli/commands/endpoint/search.py

```diff
@@ -26,6 +26,9 @@ def endpoint_search(filter_fulltext, filter_owner_id, filter_scope):
     """
     Executor for `globus endpoint search`
     """
+    if filter_scope == 'all' and not filter_fulltext:
+        raise click.UsageError('filter_scope "all" requires a search filter.')
```

Comment:
This seems better than my suggestion. Checking on how well `click` handles multiline or long messages in `UsageError`.